solutions/usaco-766.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: usaco-766
+source: USACO Gold 2017 December
+title: Barn Painting
+author: Neo Wang
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_barnpainting_gold_dec17.html)
+
+Define $\texttt{dp}[i][j]$ as the number of paintings of barn $i$ with color
+$j$. Then, our transition for a vertex $v$ where $e$ iterates over the
+subtrees of $v$ is:
```

Comment:
```suggestion
Define $\texttt{dp}[i][j]$ as the number of paintings of subtree $i$, painting barn $i$ with color $j$. Then, our transition for a vertex $v$ where $e$ iterates over the subtrees of $v$ is:
```
idk if this is better -- just wanted to mention that dp[i] is # of ways to color the **`i`th subtree** instead of just the node.